core-data/reducer.js

```diff
@@ -73,6 +73,23 @@ export function users( state = { byId: {}, queries: {} }, action ) {
 	return state;
 }
 
+/**
+ * Reducer managing taxonomies.
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export function taxonomies( state = [], action ) {
+	switch ( action.type ) {
+		case 'RECEIVE_TAXONOMIES':
+			return action.taxonomies;
+	}
+
+	return state;
+}
```

Comment:
Instead of adding a new reducer/selectors/actions, we could just leverage the new entities abstraction to automatically generate those. https://github.com/WordPress/gutenberg/blob/master/core-data/entities.js#L6

Currently, the entities only support fetching by id `getEntityRecord` but we should definitely introduce a findAll/findQuery as well `getEntityRecords( state, kind, entity, query )`

If you think it's too much work, it's fine to do it separately.

Reply:
I'd like to handle separately (and also include more tests within `core-data`). I've filed an issue #6780